H2TauTau/python/proto/analyzers/H2TauTauTreeProducerBase.py

```diff
@@ -206,6 +207,61 @@ def fillExtraMetInfo(self, tree, event):
         self.fill(tree, 'pfmet_mt1', DiTau.calcMT(event.pfmet, event.leg1))
         self.fill(tree, 'pfmet_mt2', DiTau.calcMT(event.pfmet, event.leg2))
 
+    # TauSpinner information
+    def bookTauSpinner(self, tree):
+	self.var(tree, 'TauSpinnerWTisValid')
+	self.var(tree, 'TauSpinnerWT')
+	self.var(tree, 'TauSpinnerWThminus')
+	self.var(tree, 'TauSpinnerWThplus')
+	self.var(tree, 'TauSpinnerTauPolFromZ')
+	self.var(tree, 'TauSpinnerWRight')
+	self.var(tree, 'TauSpinnerWLeft')
+	self.var(tree, 'TauSpinnerIsRightLeft')
+
+    def fillTauSpinner(self, tree, event):
+	self.fill(tree, 'TauSpinnerWTisValid', event.TauSpinnerWTisValid )
+	self.fill(tree, 'TauSpinnerWT', float(event.TauSpinnerWT))
+	self.fill(tree, 'TauSpinnerWThminus', float(event.TauSpinnerWThminus))
+	self.fill(tree, 'TauSpinnerWThplus', float(event.TauSpinnerWThplus))
+	self.fill(tree, 'TauSpinnerTauPolFromZ', float(event.TauSpinnerTauPolFromZ))
+	self.fill(tree, 'TauSpinnerWRight', float(event.TauSpinnerWRight))
+	self.fill(tree, 'TauSpinnerWLeft', float(event.TauSpinnerWLeft  ))
+	self.fill(tree, 'TauSpinnerIsRightLeft', float(event.TauSpinnerIsRightLeft ))
+
+    def bookTopPtReweighting(self, tree):
+	self.var(tree, 'gen_top_1_pt')
+        self.var(tree, 'gen_top_2_pt')
+        self.var(tree, 'gen_top_weight')
+
+    def fillTopPtReweighting(self, tree, event):
+        
+#        import pdb; pdb.set_trace()
+
+	if self.cfg_comp.isMC and self.cfg_comp.name.find('TT')!=-1 and len(event.gentopquarks)==2:
```

Comment:
from where do you get the gentopquarks? are you sure this definition is in line with what the twiki writes?